From 6936500df0872df27839acc0044bcfeeca833bf7 Mon Sep 17 00:00:00 2001
From: Mihai Serb <mihai.serb@freescale.com>
Date: Thu, 22 Mar 2012 15:33:34 +0000
Subject: [PATCH 127/518] Replace default FQID allocation mechanism

Update DPA IPSec code to work with the changes in QMan driver.
Replace calls to qm_fq_new / qm_fq_free_flags with qman_alloc_fqid /
qman_release_fqid calls.

Signed-off-by: Mihai Serb <mihai.serb@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_ipsec.c |   12 ++++--------
 1 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_ipsec.c b/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
index d37db89..17d6c21 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
@@ -1534,8 +1534,8 @@ static int get_new_fqid(struct dpa_ipsec *dpa_ipsec, uint32_t *fqid)
 	}
 
 	/* No pool defined. Get FQID from default allocator. */
-	*fqid = qm_fq_new();
-	if (*fqid == 0) {
+	err = qman_alloc_fqid(fqid);
+	if (err < 0) {
 		xx_pr_err("FQID allocation (no pool) failure.\n");
 		return -ERANGE;
 	}
@@ -1546,7 +1546,6 @@ static int get_new_fqid(struct dpa_ipsec *dpa_ipsec, uint32_t *fqid)
 static void put_free_fqid(uint32_t fqid)
 {
 	struct dpa_ipsec *dpa_ipsec;
-	int err;
 
 	/* sanity checks */
 	if (!gbl_dpa_ipsec) {
@@ -1558,11 +1557,8 @@ static void put_free_fqid(uint32_t fqid)
 	/* recycle the FQID */
 	if (dpa_ipsec->config.fqid_pool != NULL)
 		qman_fqid_pool_free(dpa_ipsec->config.fqid_pool, fqid);
-	else {
-		err = qm_fq_free_flags(fqid, 0);	/* only warn of error */
-		if (err < 0)
-			xx_pr_warn("Could not release FQID# %u\n", fqid);
-	}
+	else
+		qman_release_fqid(fqid);
 }
 
 static int wait_until_fq_empty(struct qman_fq *fq, int timeout)
-- 
1.7.5.4

